frontend/src/components/FileUploader.js

```diff
@@ -78,7 +78,34 @@ Dropzone.propTypes = {
   id: PropTypes.string.isRequired,
 };
 
-const FileTable = ({ onFileRemoved, files }) => (
+const FileTable = ({ onFileRemoved, files }) => {
+  const { isOpen, openModal, closeModal } = useModal()
+  const deleteFileModal = ({onFileRemoved, file, index}) => {
+    const onClose = () => {
+            onFileRemoved(index);
+            closeModal();
+    }
+    return (
+    <Modal
+      title={<h2>Delete File</h2>}
+      actions={
+        <>
+          <Button type="button" primary onClick={closeModal}>
+            Cancel
+          </Button>
+          <Button type="button" secondary  onClick={onClose}>
+            Delete
+          </Button>
+        </>
+      }>
+      <p>Are you sure you want to delete {file.originalFileName}?</p><p>This action cannot be undone.</p>
+    </Modal>
+  )
+    }
+
+  const ConnectedDeleteFileModal = connectModal(deleteFileModal)
+
+  return (
   <div className="files-table--container margin-top-2">
```

Comment:
I got the basic layout for the code from https://trussworks.github.io/react-uswds/?path=/docs/other-modal--full-example and the basic language from https://preview.uxpin.com/ad995fd722bd26c0f9a1c8a115f1b41833637308#/pages/134880467/simulate/no-panels?mode=i